### PR TITLE
NFDIV-2216 - AOS SOT not displayed

### PR DESCRIPTION
**Pull request by danielbe11**
### Change description ###

- To migrate Application::applicant2StatementOfTruth field to AOS::statementOfTruth.
- Always show the AOS::statementOfTruth in the AOS field.
- Only apply this in AAT for now.
- Must merge https://github.com/hmcts/cnp-flux-config/pull/15736/files first

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2216

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: charts/nfdiv-case-api/values.preview.template.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ java:
   environment:
     CASE_DATA_STORE_BASEURL: "http://ccd-data-store-api-nfdiv-case-api-pr-${CHANGE_ID}.service.core-compute-preview.internal"
     CITIZEN_UPDATE_CASE_STATE_ENABLED: true
+    SOT_MIGRATION_ENABLED: true
```
Comment (velimd) [inline charts/nfdiv-case-api/values.preview.template.yaml:11]: is this needed when you set it to true in jenkinsfile or vice versa?

File: charts/nfdiv-case-api/values.yaml [hunk 2]
```diff
@@ -54,6 +54,7 @@ java:
     BULK_ACTION_BATCH_SIZE_MIN: 30
     BULK_ACTION_BATCH_SIZE_MAX: 50
     CITIZEN_UPDATE_CASE_STATE_ENABLED: false
+    SOT_MIGRATION_ENABLED: false
```
Comment (velimd) [inline charts/nfdiv-case-api/values.yaml:57]: similar here as mentioned above.

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java [hunk 3]
```diff
@@ -57,8 +57,4 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
             .state(state)
             .build();
     }
-
-    public boolean isCitizenUpdateCaseStateEnabled() {
```
Comment (velimd) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenUpdateCaseStateAat.java:61]: why is this removed?

**Comment by velimd** [comment]
other than the conversation we had, its LGTM

**Review by velimd — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/RetiredFields.java [hunk 4]
```diff
@@ -91,4 +98,10 @@ public static int getVersion() {
     private static TriConsumer<Map<String, Object>, String, Object> moveTo(String newFieldName) {
         return (data, key, val) -> data.put(newFieldName, val);
     }
+
+    private static boolean shouldMigrateSOT(Map<String, Object> data) {
+        return Boolean.parseBoolean(System.getenv().get("SOT_MIGRATION_ENABLED"))
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/RetiredFields.java:103]: better to use ` Boolean.parseBoolean(System.getenv().getOrDefault("SOT_MIGRATION_ENABLED","false"))`

Comment (danielbe11) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/RetiredFields.java]: Done

**Review by nitinprabhu — APPROVED** [review]
